src/Compilers/Core/Portable/Compilation/OperationExtensions.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Immutable;
+using System.Collections.Generic;
+using System.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    internal interface IOperationSearchable
+    {
+        IEnumerable<IOperation> Descendants();
+        IEnumerable<IOperation> DescendantsAndSelf();
+    }
+
+    public static class OperationExtensions
+    {
+        public static IEnumerable<IOperation> Descendants(this IOperation operation)
+        {
+            var searchable = operation as IOperationSearchable;
+            if (searchable != null)
+            {
+                return searchable.Descendants();
+            }
+            else
+            {
+                return SpecializedCollections.EmptyEnumerable<IOperation>();
+            }
+        }
+
+        public static IEnumerable<IOperation> DescendantsAndSelf(this IOperation operation)
+        {
+            var searchable = operation as IOperationSearchable;
+            if (searchable != null)
+            {
+                return searchable.DescendantsAndSelf();
+            }
+            else
+            {
+                return SpecializedCollections.EmptyEnumerable<IOperation>();
+            }
+        }
+
+        public static IOperation GetRootOperation(this ISymbol symbol, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var symbolWithOperation = symbol as ISymbolWithOperation;
+            if (symbolWithOperation != null)
+            {
+                return symbolWithOperation.GetRootOperation(cancellationToken);
+            }
+            else
+            {
+                return null;
+            }
+        }
+    }
+
+    internal interface ISymbolWithOperation
```

Comment:
What is an example of an ISymbolWithOperation and where would this be used? Would the body of a MethodSymbol be the root operation for the method? If so, would an analyzer need to get the body of the method directly?